### PR TITLE
fix: CUD normalization

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change

(A few sentences about this PR)

## Related issues

- Link to issue1 here
- Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your
changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here
highlighting the necessary changes)

## Checklist for important updates

- [ ] Changelog has been updated
    - [ ] If there are any db schema changes, mention those changes clearly
- [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
- [ ] `pluginInterfaceSupported.json` file has been updated (if needed)
- [ ] Changes to the version if needed
    - In `build.gradle`
- [ ] If added a new paid feature, edit the `getPaidFeatureStats` function in FeatureFlag.java file
- [ ] Had installed and ran the pre-commit hook
- [ ] If there are new dependencies that have been added in `build.gradle`, please make sure to add them
  in `implementationDependencies.json`.
- [ ] Issue this PR against the latest non released version branch.
    - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the
      latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    - If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR

- [ ] Item1
- [ ] Item2

### CODE REVIEW

File: src/main/java/io/supertokens/webserver/Utils.java [hunk 1]
```diff
@@ -41,10 +41,17 @@ public static String normalizeAndValidateConnectionUriDomain(String connectionUr
             throw new ServletException(new WebserverAPI.BadRequestException("connectionUriDomain should not be an empty String"));
         }
 
-        if (!connectionUriDomain.matches("^[a-z0-9-]+(\\.[a-z0-9-]+)*(:[0-9]+)?$")) {
+        String hostnameRegex = "^[a-z][a-z0-9-]+(\\.[a-z][a-z0-9-]+)*(:[0-9]+)?$";
+        String ipRegex = "^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(:[0-9]+)?$";
+
+        if (!connectionUriDomain.matches(hostnameRegex) && !connectionUriDomain.matches(ipRegex)) {
             throw new ServletException(new WebserverAPI.BadRequestException("connectionUriDomain is invalid"));
         }
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/webserver/Utils.java:50]: use a url parsing library

File: src/main/java/io/supertokens/webserver/WebserverAPI.java [hunk 2]
```diff
@@ -272,7 +272,13 @@ private String getAppId(HttpServletRequest req) {
     }
 
     private String getConnectionUriDomain(HttpServletRequest req) {
-        String connectionUriDomain = req.getServerName() + ":" + req.getServerPort();
+        String connectionUriDomain = req.getServerName();
+        try {
+            connectionUriDomain = Utils.normalizeAndValidateConnectionUriDomain(connectionUriDomain);
+        } catch (ServletException e) {
+            throw new IllegalStateException("should never come here");
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/webserver/WebserverAPI.java:279]: just throw the ServletException

File: src/test/java/io/supertokens/test/TestingProcessManager.java [hunk 3]
```diff
@@ -129,10 +129,7 @@ public void kill(boolean removeAllInfo) throws InterruptedException {
             if (killed) {
                 return;
             }
-            // we check if there are multiple user pool IDs loaded, and if there are,
-            // we clear all the info before killing cause otherwise those extra dbs will retain info
-            // across tests
-            if (removeAllInfo && StorageLayer.hasMultipleUserPools(this.main)) {
```
Comment (rishabhpoddar) [inline src/test/java/io/supertokens/test/TestingProcessManager.java:135]: remove this from plugins also

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/webserver/Utils.java (hunk 1 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/webserver/Utils.java:52]: Done

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**